### PR TITLE
DROOLS-4972: Buttons stay disabled when switching to textual collection editor

### PR DESCRIPTION
**Pull request by yesamer**
@danielezonca @jomarko @dupliaka 

https://issues.redhat.com/browse/DROOLS-4972

### CODE REVIEW

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/collectioneditor/CollectionView.java [hunk 1]
```diff
@@ -147,13 +151,11 @@
...
-    ButtonElement getCancelButton();
-
-    ButtonElement getRemoveButton();
-
-    ButtonElement getSaveButton();
+    /**
+     * It enables/disables buttons group which manage the Collection Editor
+     * @param enable
+     */
+    void disableCollectionEditorButtons(boolean enable);
```
Comment (dupliaka) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/collectioneditor/CollectionView.java]: I feel that confusing-parameter name should be disable according to the function name.
Enable of disableness is hard to imagine)

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/collectioneditor/CollectionPresenter.java [hunk 2]
```diff
@@ -70,6 +70,11 @@
      */
     protected Map<String, Map<String, Map<String, String>>> expandablePropertiesMap = new HashMap<>();
 
+    /**
+     * It defines if <b>Create collection</b> guided editor is in editing status or not.
+     */
+    protected boolean createCollectionIsEditing = false;
```
Comment (danielezonca) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/collectioneditor/CollectionPresenter.java]: What about define an enum to keep the status of the whole panel? Something like `GUIDED`, `EXPRESSION` and `GUIDED_ADDING_ELEMENT`. I think it is easier to understand

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/collectioneditor/CollectionView.java (hunk 1 above)
Comment (yesamer) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/collectioneditor/CollectionView.java]: @dupliaka You're right, I forgot to update the name.

**Comment by yesamer** [comment]
@danielezonca @dupliaka I changed the fix, now It simply disables the "Define Collection" radio button when Editing mode is active. This simplify the logic and the change (a status is no longer required). Thank you for your suggestions.

**Comment by yesamer** [comment]
@kiegroup/gatekeepers Can your please merge it?

**Pull request merged**